Godot 3 with Mono compilation error: expected type-specifier before 'AABB'

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Ubuntu 17.04 x64
g++ (Ubuntu 6.3.0-12ubuntu2) 6.3.0 20170406

**Issue description:**
I'm trying to compile Godot 3 with Mono support (calling scons with flags `module_mono_enabled=yes` and `mono_glue=no`) and compilation fails with next error:

> modules/mono/mono_gd/gd_mono_marshal.cpp: In function 'MonoObject* GDMonoMarshal::variant_to_mono_object(const Variant*, const ManagedType&)':
> modules/mono/mono_gd/gd_mono_marshal.cpp:301:25: error: expected type-specifier before 'AABB'
>      RETURN_BOXED_STRUCT(AABB, p_var);
>                          ^
> modules/mono/mono_gd/gd_mono_marshal.cpp:39:40: note: in definition of macro 'RETURN_BOXED_STRUCT'
>    const m_t &m_in = m_var_in->operator m_t();                           \
>                                         ^~~
> modules/mono/mono_gd/gd_mono_marshal.cpp:398:26: error: expected type-specifier before 'AABB'
>       RETURN_BOXED_STRUCT(AABB, p_var);
>                           ^
> modules/mono/mono_gd/gd_mono_marshal.cpp:39:40: note: in definition of macro 'RETURN_BOXED_STRUCT'
>    const m_t &m_in = m_var_in->operator m_t();                           \
>                                         ^~~
> scons: *** [modules/mono/mono_gd/gd_mono_marshal.x11.tools.64.mono.o] Error 1
> scons: building terminated because of errors.
> 

**Steps to reproduce:**
1. Clone this repository at branch `master` (reproduced at commit 63283eca5542314ea5b71afa1fe39bd48c202a4d)
2. Go to `godot` folder and call
`scons platform=x11 tools=yes module_mono_enabled=yes mono_glue=no`
3. Notice that compilation fails.

What can I do with that?

--- a/modules/mono/mono_gd/gd_mono_field.cpp
+++ b/modules/mono/mono_gd/gd_mono_field.cpp
@@ -41,7 +41,7 @@ void GDMonoField::set_value_raw(MonoObject *p_object, void *p_ptr) {
 void GDMonoField::set_value(MonoObject *p_object, const Variant &p_value) {
 #define SET_FROM_STRUCT_AND_BREAK(m_type)                \
 	{                                                    \
-		const m_type &val = p_value.operator m_type();   \
+		const m_type &val = p_value.operator ::m_type(); \
 		MARSHALLED_OUT(m_type, val, raw);                \
 		mono_field_set_value(p_object, mono_field, raw); \
 		break;                                           \

--- a/modules/mono/mono_gd/gd_mono_marshal.cpp
+++ b/modules/mono/mono_gd/gd_mono_marshal.cpp
@@ -36,7 +36,7 @@ namespace GDMonoMarshal {
 
 #define RETURN_BOXED_STRUCT(m_t, m_var_in)                                    \
 	{                                                                         \
-		const m_t &m_in = m_var_in->operator m_t();                           \
+		const m_t &m_in = m_var_in->operator ::m_t();                         \
 		MARSHALLED_OUT(m_t, m_in, raw);                                       \
 		return mono_value_box(mono_domain_get(), CACHED_CLASS_RAW(m_t), raw); \
 	}

